pyglmnet/pyglmnet.py

```diff
@@ -331,6 +338,28 @@ def grad_L2loss(self, beta0, beta, reg_lambda, X, y):
 
         return grad_beta0, grad_beta
 
+    def grad_(self, beta0, beta, reg_lambda, X, y):
+        """Concatenates the intercept gradient with the feature gradient"""
+        grad_beta0, grad_beta = self.grad_L2loss(beta0, beta, reg_lambda, X, y)
+        return np.concatenate(grad_beta0, grad_beta)
+
+    def grad_hess_(self, beta0, beta, reg_lambda, X, y):
+        """
+        A callable function passed to optimize.newton_cg
+        Returns gradient and a callable for hessian vector product
+        """
+        g = self.grad_(beta0, beta, reg_lambda, X, y)
+
+        def hess_product(v):
```

Comment:
@jasmainak yes, aware of the convention, but the way sklearn has implemented this (see example linked above), `hess_product()` is a callable returned by `grad_hess_()` and needs to be nested :)